components/event-bus/internal/knative/subscription/controller/knativesubscription/controller.go

```diff
@@ -1,47 +1,45 @@
 package knativesubscription
 
 import (
-	evapisv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	subscriptioninformersv1alpha1 "knative.dev/eventing/pkg/client/injection/informers/messaging/v1alpha1/subscription"
+	"knative.dev/eventing/pkg/reconciler"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+
+	eventbusscheme "github.com/kyma-project/kyma/components/event-bus/client/generated/clientset/internalclientset/scheme"
 	"github.com/kyma-project/kyma/components/event-bus/internal/knative/util"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"sigs.k8s.io/controller-runtime/pkg/handler"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logf.Log.WithName("knative-subscription-controller")
-
 const (
+	// reconcilerName is the name of the reconciler
+	reconcilerName = "KnativeSubscriptions"
+
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
 	controllerAgentName = "knative-subscription-controller"
 )
 
-// ProvideController instantiates a reconciler which reconciles Knative Subscriptions.
-func ProvideController(mgr manager.Manager) error {
+func init() {
+	// Add sources types to the default Kubernetes Scheme so Events can be
+	// logged for sources types.
+	runtime.Must(eventbusscheme.AddToScheme(scheme.Scheme))
```

Comment:
This controller records events for _Knative_ Subscriptions, so we also need to register Knative types.
```golang
// import kneventingscheme "knative.dev/eventing/pkg/client/clientset/versioned/scheme"
runtime.Must(kneventingscheme.AddToScheme(scheme.Scheme))
```

Reply:
@antoineco and the import `	knativescheme "knative.dev/eventing/pkg/client/clientset
/versioned/scheme"`
?